src/Joy4004482.cpp

```diff
@@ -19,18 +19,16 @@
 #include <stdexcept>
 #include "Joy4004482.h"
 //---------------------------------------------------------------------------
-Joy4004482::Joy4004482(IifGPIO *pio, TSettings::SetJoystick* settings)
+Joy4004482::Joy4004482(IifGPIO *pio, TSettings::SetJoystick* settings) :
+	joy{{ nullptr, nullptr, false, 0 }, { nullptr, nullptr, false, 0 }},
+	joyCnt(0),
+	sameDev(false)
 {
 	if (!pio)
 		throw std::runtime_error("Required GPIO not initialized!");
 
 	this->pio = pio;
 	this->settings = settings;
```

Comment:
Do you think, that we could set `pio` and `settings` same way?

So that throw (plus include) we could omit...? 🤔

Reply:
yes, I think we can reduce the constructor code to just initializer
```c++
Joy4004482::Joy4004482(IifGPIO *pio, TSettings::SetJoystick* settings) :
	pio(pio),
	settings(settings),
	joy{{ NULL, NULL, false, 0 }, { NULL, NULL, false, 0 }},
	joyCnt(0),
	sameDev(false)
{
}
```